README.md

```diff
@@ -546,7 +547,17 @@ To contribute to the gem, please clone the repo and cut a new branch. In the PR
     *Your Name*
 ```
 
-Please **do not** bump the gem version in your PR. This will be done in a follow up PR by the gem maintainers.
+Please **do not** bump the gem version in your PR. This will be done in a follow up by the gem maintainers.
+
+## Releasing
+
+Gem maintainer should follow these steps to release a new version:
+
+1. Merge all PRs into the `master` branch that you want included in the release.
```

Comment:
Thoughts about using the native Github Releases functionality? This would remove a few steps below and make the changelog not necessary... We did it for AutoAPI (and a couple others) – I think it looks nice :) https://github.com/procore/autoapi/releases

Reply:
Which steps would that remove? CircleCI still needs to see a tagged commit to trigger the publishing job.

Having a changelog file is useful because it allows contributors to write the changelog entry in their PR.

Reply:
This sounds interesting. Do you have more information on this?

@m-graf Is this a good PR to look at https://github.com/procore/autoapi/pull/267

Reply:
> Which steps would that remove? CircleCI still needs to see a tagged commit to trigger the publishing job.
> 
> Having a changelog file is useful because it allows contributors to write the changelog entry in their PR.

Step 2 can be removed (versions can be created, tagged, and retrieved programmatically from GitHub's API, so no more hardcoding these values)
Step 3-4 are moved to the UI (see /releases)
Step 5 can also be removed (again, will require changes to the CI process, but can and [should?] be standardized).



> This sounds interesting. Do you have more information on this?
> 
> @m-graf Is this a good PR to look at [procore/autoapi#267](https://github.com/procore/autoapi/pull/267)

Yes. This was / still is within our locus of control... switching over to the actual tagging / release functionality (which comes with Autogenerated release notes!) is super simple... On top of that, modifying the behavior of CI is also pretty straight forward... In a past live I used the CI to write the API's version to an object store that would be retrieved by a `status` endpoint. It was nice because new versions would get automatically updated, and performing a `get` on the endpoint would yield something like this:

```
{
  "status": "ok",
  "name": "something-something-api",
  "version": "1.0.58336"
}
```